### PR TITLE
Exposing the lower and upper limits

### PR DESCRIPTION
**Pull request by buyology**
Please accept this PR that exposes the lower and upper limits as settings to the user

Tried to follow the behaviour of the R wrapper:
* either input an array or a single value of lower and upper limits
* check that lower_limits <= 0 and upper_limits >= 0

**Testing done:**
* Added unit tests

### CODE REVIEW

**Comment by rajhans** [comment]
This change is super useful. Any ETA on accepting this change?
@wlattner

File: glmnet/linear.py [hunk 1]
```diff
@@ -137,6 +149,13 @@ def __init__(self, alpha=1, n_lambda=100, min_lambda_ratio=1e-4,
         self.random_state = random_state
         self.verbose = verbose
 
+        if any(self.lower_limits)>0 if isinstance(self.lower_limits, np.ndarray) else self.lower_limits>0:
+            raise ValueError("lower_limits must be non-positive")
+
+        if any(self.upper_limits)<0 if isinstance(self.upper_limits, np.ndarray) else self.upper_limits<0:
+            raise ValueError("upper_limits must be positive")
```
Comment (jseabold) [inline glmnet/linear.py]: Do you mind checking the line lengths and adding spaces around the operators here?

File: glmnet/linear.py [hunk 2]
```diff
@@ -46,6 +46,15 @@ class ElasticNet(BaseEstimator):
     fit_intercept : bool, default True
         Include an intercept term in the model.
 
+    lower_limits : array, (shape n_features,) default -infinity
+        Array of lower limits for each coefficient, must be non-postive.
```
Comment (jseabold) [inline glmnet/linear.py]: Typo in positive.

File: glmnet/logistic.py [hunk 3]
```diff
@@ -48,6 +48,15 @@ class LogitNet(BaseEstimator):
     fit_intercept : bool, default True
         Include an intercept term in the model.
 
+    lower_limits : array, (shape n_features,) default -infinity
+        Array of lower limits for each coefficient, must be non-postive.
```
Comment (jseabold) [inline glmnet/logistic.py]: Typo.

File: glmnet/logistic.py [hunk 4]
```diff
@@ -142,6 +154,13 @@ def __init__(self, alpha=1, n_lambda=100, min_lambda_ratio=1e-4,
         self.random_state = random_state
         self.verbose = verbose
 
+        if any(self.lower_limits)>0 if isinstance(self.lower_limits, np.ndarray) else self.lower_limits>0:
+            raise ValueError("lower_limits must be non-positive")
```
Comment (jseabold) [inline glmnet/logistic.py]: Same style comment.

**Comment by melvinev** [comment]
Any chance this will be merged?

**Comment by stephen-hoover** [comment]
@melvinev , it looks like the original PR writer hasn't had time to finish this up. If you'd like to open a new PR with these changes, we can review it and get it merged.

**Comment by buyology** [comment]
Great, thanks for picking this up. Totally slipped my mind :)

**Pull request closed without merging**